cobalt/persistent_storage/persistent_settings.cc

```diff
@@ -164,70 +171,74 @@ PersistentSettings::GetPersistentSettingAsDictionary(const std::string& key) {
   return dict;
 }
 
+void PersistentSettings::PostBlockingTask() {
```

Comment:
How is this different than `base::task_runner_util::WaitForFence(task_runner(), FROM_HERE)`? It's technically even a little worse, because it will lose the call site information due to thethe `FROM_HERE` being evaluated in the helper function.

Reply:
Using `WaitForFence()` fails tests due to the `DCHECK(!task_runner->RunsTasksInCurrentSequence())` https://source.corp.google.com/h/lbshell-internal/cobalt_src/+/COBALT:cobalt/base/task_runner_util.cc;l=50

Reply:
That DCHECK fails when you're trying to post a task to a taskrunner that runs tasks in the current sequence, the translation for us here is that you're already running in the same thread that is supposed to run the task, so you'll deadlock waiting for the task to finish because the task runner is waiting for your code to return...

Reply:
I need to take another look but that doesn't seem quite right since I'm able to run the tests without any deadlocking.

Reply:
Switched to using `WaitForFence()` with no issues now. I think it was previously complaining because of that whole closure mess that I've since removed.